flink-connector-opensearch-e2e-tests/pom.xml

```diff
@@ -0,0 +1,190 @@
+			<type>test-jar</type>
+			<exclusions>
+				<!-- Prevent this from interfering with dependency tree -->
+				<exclusion>
+					<groupId>*</groupId>
+					<artifactId>*</artifactId>
+				</exclusion>
+			</exclusions>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.logging.log4j</groupId>
+			<artifactId>log4j-api</artifactId>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-test-utils</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-test-utils</artifactId>
+			<version>${flink.version}</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.opensearch</groupId>
+			<artifactId>opensearch-testcontainers</artifactId>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-deploy-plugin</artifactId>
+				<configuration>
+					<skip>true</skip>
+				</configuration>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<finalName>opensearch-end-to-end-test</finalName>
+							<outputDirectory>dependencies</outputDirectory>
+							<artifactSet>
+								<excludes combine.children="append">
+									<exclude>com.google.code.findbugs:jsr305</exclude>
+									<exclude>org.slf4j:slf4j-api</exclude>
+								</excludes>
+							</artifactSet>
+							<filters combine.children="append">
+								<filter>
+									<artifact>*:*</artifact>
+									<excludes>
+										<exclude>META-INF/*.SF</exclude>
+										<exclude>META-INF/*.DSA</exclude>
+										<exclude>META-INF/*.RSA</exclude>
+									</excludes>
+								</filter>
+							</filters>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-dependency-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>copy</id>
+						<phase>pre-integration-test</phase>
+						<goals>
+							<goal>copy</goal>
+						</goals>
+						<configuration>
+							<artifactItems>
+								<artifactItem>
+									<groupId>org.apache.flink</groupId>
+									<artifactId>flink-connector-test-utils</artifactId>
+									<version>${flink.version}</version>
+									<destFileName>flink-connector-test-utils.jar</destFileName>
+									<type>jar</type>
+									<outputDirectory>${project.build.directory}/dependencies</outputDirectory>
+								</artifactItem>
+							</artifactItems>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+		</plugins>
+	</build>
+
+	<profiles>
+		<profile>
+			<id>run-end-to-end-tests</id>
```

Comment:
There is no longer a real benefit in not treating the e2e tests as normal IT Cases.
(notably because we always run them on CI and running them locally works just fine as well)

Reply:
@zentol this it not correct, the e2e tests need Flink distribution to be downloaded and provided, this preparation is done within CI workflows, otherwise the tests won't even start.

Reply:
That's a good point.

We should fix that though (in a follow-up; across connectors); the tests should be able to download Flink themselves. These magical prerequisite steps just make it difficult to test things locally.

Reply:
:+1: second that. It is, took me a while to understand how I could run it locally